test/functional/test_framework/test_node.py

```diff
@@ -297,9 +297,21 @@ def wait_for_cookie_credentials(self):
             time.sleep(1.0 / poll_per_s)
         self._raise_assertion_error("Unable to retrieve cookie credentials after {}s".format(self.rpc_timeout))
 
-    def generate(self, nblocks, maxtries=1000000):
+    def generate(self, nblocks, maxtries=1000000, **kwargs):
         self.log.debug("TestNode.generate() dispatches `generate` call to `generatetoaddress`")
-        return self.generatetoaddress(nblocks=nblocks, address=self.get_deterministic_priv_key().address, maxtries=maxtries)
+        return self.generatetoaddress(nblocks=nblocks, address=self.get_deterministic_priv_key().address, maxtries=maxtries, **kwargs)
+
+    def generateblock(self, *args, invalid_call, **kwargs):
```

Comment:
This `invalid_call` argument seems a little bit hand-holdy. We're all adults here - can we just leave a comment saying don't call this from the individual tests?

Reply:
I don't think leaving a comment is enough. How do you ensure everyone reads the comment? Let's continue the discussion in https://github.com/bitcoin/bitcoin/pull/23207 , though.